app/controllers/financials_controller.rb

```diff
@@ -0,0 +1,22 @@
+class FinancialsController < ApplicationController
+  def show
+    @expenses = get_table("expenses")
+    @total_expenses = @expenses["data"].sum { |e| e["total"] }
+    @revenue = get_table("revenue")
+    @total_revenue = @revenue["data"].sum { |e| e["total"] }
+    @overview = overview_data
+    @total = @total_revenue - @total_expenses
+  end
+
+  private
+
+  def overview_data
+    @expenses["data"].group_by { |e| e["date"].to_date.strftime("%B, %Y") }.merge(@revenue["data"].group_by { |e| e["date"].to_date.strftime("%B, %Y") }).map { |e, i|
+      {month: e, expense_total: (i.sum { |y| (y["description"] ? y["total"] : 0) }), revenue_total: (i.sum { |y| (y["source"] ? y["total"] : 0) })}
+    }.map { |e| {month: e[:month], expense_total: e[:expense_total], revenue_total: e[:revenue_total], total: (e[:revenue_total] - e[:expense_total])} }
+  end
+
+  def get_table(table)
+    YAML.safe_load(File.read("#{Rails.root}/data/financials.yml"))["financials"][table]
+  end
+end
```

Comment:
Let's move all of this to a model so the controller doesn't have to do any work.

```
class IncomeReport
  def expenses
     # ...
  end

  def total_expenses
    # ...
  end
end
```

Then, in the controller, `@report = IncomeReport.new`. And the view can `@report.expenses`.

I'd also love to move all magic strings and symbols to that new model. I don't like that the view/controller knows about the structure of the YAML files. This should all be abstracted away in the `IncomeReport` model.

Finally, once all the logic is in a PORO we can add tests!